content/src/main/content/jcr_root/apps/acs-commons/gui/components/authoring/dialogshared/dialogshared.jsp

```diff
@@ -0,0 +1,20 @@
+<%@ page import="javax.jcr.Session" %>
+<%@ page import="com.day.cq.commons.jcr.JcrUtil" %>
+<%@page session="false"%>
+<%@taglib prefix="sling" uri="http://sling.apache.org/taglibs/sling" %>
+
+<sling:defineObjects />
+
+<%
+    String dialogSharedDataPath = slingRequest.getRequestPathInfo().getSuffix();
+    Session dialogSharedSession = resourceResolver.adaptTo(Session.class);
+
+    // Ensure the path exists in the JCR so that we don't get a NPE
+    // when attempting to load the dialog.
+    if (!dialogSharedSession.nodeExists(dialogSharedDataPath)) {
+        JcrUtil.createPath(dialogSharedDataPath, "nt:unstructured", "nt:unstructured", dialogSharedSession, false);
```

Comment:
The issue is that the OOTB AEM dialog code pukes if the node is multiple levels deep and the nodes in between dont yet exist.

SharedComponentPropsPageInfoProvider checks if the current user has permissions to write that node, and if not then flags the shared component properties feature as disabled, which in turn makes it so the UI does not give the user the buttons to configure shared/global configs (and thus they will not hit this code).